File: ruby/acronym/acronym.rb
module Acronym

  VERSION = 1

  def self.abbreviate(name)
    words(name).map do |word|
      acr = word.scan(/[A-Z]/)
      acr.empty? || acr.length == word.length ? word.chr.upcase : acr
    end.join
  end

  def self.words(name)
    name.scan(/\w+/)
  end

end

File: elixir/acronym/acronym.exs
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    string
    |> String.split
    |> Enum.map(&extract(&1))
    |> Enum.join
  end

  defp extract(word) do
    [String.first(word) | Regex.scan(~r/[A-Z]/, String.slice(word, 1..-1))]
    |> Enum.join
    |> String.upcase
  end
end
